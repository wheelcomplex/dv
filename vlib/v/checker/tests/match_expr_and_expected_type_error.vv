ch := `a`
match ch {
  'a' {}
  else {}
}

char := 123
match char {
  'a' {}
  else {}
}
