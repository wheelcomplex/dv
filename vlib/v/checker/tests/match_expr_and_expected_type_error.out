vlib/v/checker/tests/match_expr_and_expected_type_error.vv:2:1: error: cannot use type `rune` as type `string`
    1 | ch := `a`
    2 | match ch {
      | ~~~~~~~~~~
    3 |   'a' {}
    4 |   else {}
vlib/v/checker/tests/match_expr_and_expected_type_error.vv:8:1: error: cannot use type `int` as type `string`
    6 | 
    7 | char := 123
    8 | match char {
      | ~~~~~~~~~~~~
    9 |   'a' {}
   10 |   else {}
